server/src/main/java/org/opensearch/tasks/TaskInfo.java

```diff
@@ -308,6 +358,10 @@ public static TaskInfo fromXContent(XContentParser parser) {
         }
         @SuppressWarnings("unchecked")
         TaskResourceStats resourceStats = (TaskResourceStats) a[i++];
+        long cancelledAt = -1;
+        if (cancelled) {
+            cancelledAt = (Long) a[i++];
+        }
```

Comment:
I believe we still need to read from the parsed object; whether or not `cancelled` is true. Otherwise, the counter won't be incremented (`a[i++]`) and decoding might fail for new fields that might be added in the future.

Reply:
Hmm make sense. Changed.